instrumentation/spring/spring-scheduling-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/scheduling/v3_1/TaskSchedulerInstrumentation.java

```diff
@@ -20,7 +19,10 @@
 public class TaskSchedulerInstrumentation implements TypeInstrumentation {
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return implementsInterface(named("org.springframework.scheduling.TaskScheduler"));
+    return namedOneOf(
+        "org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler",
+        "org.springframework.scheduling.concurrent.ConcurrentTaskScheduler",
+        "org.springframework.scheduling.commonj.TimerManagerTaskScheduler");
```

Comment:
can you add a comment, to explain why we aren't instrumenting all TaskSchedulers? (or do we think this is a good general pattern and can skip comment? i'm ok with that too)

Reply:
Added 👍